it/uk/gov/hmrc/incorporatedentityidentificationfrontend/controllers/SupportWelshTranslationControllerISpec.scala

```diff
@@ -0,0 +1,156 @@
+  with AuthStub {
+
+  "all the views requested via GET" should {
+
+    object TestPrecondition {
+      type TestPrecondition = () => Any
+      val noNeeded: TestPrecondition = () => ()
+      val checkYourAnswers: TestPrecondition = () => stubRetrieveIncorporatedEntityInformation(testJourneyId)(OK, testIncorporatedEntityFullJourneyDataJson)
+      val confirmCompanyName: TestPrecondition = () => stubRetrieveCompanyProfileFromBE(testJourneyId)(status = OK, body = Json.toJsObject(testCompanyProfile))
+      val confirmCtutr: TestPrecondition = () => stubRetrieveCtutr(testJourneyId)(status = OK, body = testCtutr)
+      val confirmChrn: TestPrecondition = () => stubRetrieveChrn(testJourneyId)(status = OK, body = testCHRN)
+      val confirmBusinessName: TestPrecondition = () => noNeeded
+
+    }
+
+    val allGETUrlsToBeTested: TableFor2[String, TestPrecondition.TestPrecondition] =
+      Tables.Table(
+        ("urlToBeTested", "doThisToCreateTestPrecondition"),
+        (s"$baseUrl/$testJourneyId/company-number", TestPrecondition.confirmCompanyName),
+        (s"$baseUrl/$testJourneyId/check-your-answers-business", TestPrecondition.checkYourAnswers),
+        (s"$baseUrl/$testJourneyId/ct-utr", TestPrecondition.confirmCtutr),
+        (s"$baseUrl/$testJourneyId/confirm-business-name", TestPrecondition.confirmBusinessName),
+        (s"$baseUrl/$testJourneyId/chrn", TestPrecondition.confirmChrn)
+      )
+
+    "display welsh translation when cy cookie is specified" in {
+      await(journeyConfigRepository.insertJourneyConfig(
+        journeyId = testJourneyId,
+        authInternalId = testInternalId,
+        journeyConfig = testDefaultWelshJourneyConfig
+      ))
+
+      stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+
+      forAll(allGETUrlsToBeTested) { (getUrlToBeTested, doThisToCreateTestPrecondition) =>
+
+        lazy val actualDocFromResponse: Document = {
+          doThisToCreateTestPrecondition()
+          extractDocumentFrom(aWSResponse = get(getUrlToBeTested, cookie = cyLangCookie))
+        }
+
+        actualDocFromResponse.getServiceName.text mustBe testDefaultWelshJourneyConfig.pageConfig.optLabels.get.optWelshServiceName
+
+      }
+    }
+
+    "display the custom welsh translation in the pageConfig when cy cookie is specified" in {
+      await(journeyConfigRepository.insertJourneyConfig(
+        journeyId = testJourneyId,
+        authInternalId = testInternalId,
+        journeyConfig = testDefaultWelshJourneyConfig
+      ))
+
+      stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+
+      forAll(allGETUrlsToBeTested) { (getUrlToBeTested, doThisToCreateTestPrecondition) =>
+
+        lazy val actualDocFromResponse: Document = {
+          doThisToCreateTestPrecondition()
+          extractDocumentFrom(aWSResponse = get(getUrlToBeTested, cookie = cyLangCookie))
+        }
+
+        actualDocFromResponse.getServiceName.text mustBe testWelshServiceName
```

Comment:
And the result will then be the value defined in the optLabels property of the config above.